crates/typst/src/model/table.rs

```diff
@@ -185,8 +185,11 @@ impl Layout for TableElem {
             styles,
             self.span(),
         );
-
-        Ok(layouter.layout(engine)?.fragment)
+        let mut frag = layouter.layout(engine)?.fragment;
+        for frame in &mut frag {
+            frame.meta(styles, false);
+        }
+        Ok(frag)
```

Comment:
Does this problem also affect grid, list and enum? (Please add tests for those!)

If so, this should probably be done inside `GridLayouter` itself, used by all 4 (table, grid, list, enum). Before doing that, though, please rebase first to avoid problems :p

If Laurenz is okay with this change, I can help here if needed.